src/context/category-context.js

```diff
@@ -0,0 +1,37 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+const CategoryContext = createContext();
+
+const CategoryProvider = ({ children }) => {
+  const [categories, setCategories] = useState([]);
+
+  const getCategories = async () => {
+    let response = [];
+    try {
+      response = await axios.get("/api/categories");
+    } catch (error) {
+      console.log(error);
```

Comment:
you can use toast to show warning/error to user instead of console.log

Reply:
Will surely work on error handling and feedback soon